shopfloor_packing_info/hooks.py

```diff
@@ -0,0 +1,131 @@
+            SET shopfloor_packing_info_tmp=shopfloor_packing_info;"""
+        )
+        cr.execute(
+            """
+            DELETE FROM ir_ui_view WHERE id IN (
+              SELECT v.id
+              FROM ir_ui_view v, ir_model_data m
+              WHERE v.arch_db ILIKE '%shopfloor_packing_info%'
+                AND m.res_id = v.id
+                AND m.module = 'shopfloor'
+                AND m.model = 'ir.ui.view');
+            """
+        )
+
+
+def post_init_hook(cr, registry):
+    if _column_exists(cr, "res_partner", "shopfloor_packing_info_tmp"):
+        copy_into_new_model(cr)
+        cr.execute(
+            """
+            ALTER TABLE res_partner
+            DROP COLUMN shopfloor_packing_info_tmp;"""
+        )
+
+
+def _column_exists(cr, table_name, column_name):
+    # See https://stackoverflow.com/a/9991073
+    cr.execute(
+        """
+        SELECT count(attname)
+        FROM pg_attribute
+        WHERE attrelid = ( SELECT oid FROM pg_class WHERE relname = %s )
+          AND attname = %s;""",
+        (table_name, column_name),
+    )
+    return cr.fetchone()[0] == 1
+
+
+def copy_into_new_model(cr):
+    env = Environment(cr, SUPERUSER_ID, {})
+    shopfloor_packing_info = env["shopfloor.packing.info"]
+    res_partner = env["res.partner"]
+    ir_model_data = env["ir.model.data"]
+
+    env.cr.execute(
+        """
+        SELECT id, COALESCE(shopfloor_packing_info_tmp, '') AS packing_info_old
+        FROM res_partner
+        WHERE active = TRUE;
+        """
+    )
+
+    for row in env.cr.dictfetchall():
+        partner_id = int(row["id"])
+        packing_info_full = row["packing_info_old"].strip()
+        if packing_info_full:
+            partner = res_partner.browse(partner_id)
+
+            # We assume the name is the first line, if there are several.
+            linesep_pos = packing_info_full.find(os.linesep)
+            if linesep_pos == -1:
+                packing_info_name = packing_info_full
+            else:
+                packing_info_name = packing_info_full[:linesep_pos]
+
+            # If a record already exists, assign it; otherwise: create and assign.
+            packing_info_record = shopfloor_packing_info.search(
+                [("name", "=", packing_info_name), ("text", "=", packing_info_full)],
+                limit=1,
+            )
+            if packing_info_record:
+                partner.shopfloor_packing_info_id = packing_info_record
+            else:
+                packing_info_record = shopfloor_packing_info.create(
+                    {"name": packing_info_name, "text": packing_info_full}
+                )
+                partner.shopfloor_packing_info_id = packing_info_record
+                xml_id = hashlib.md5(packing_info_full.encode("utf-8")).hexdigest()
+                ir_model_data._update_xmlids(
+                    [
+                        {
+                            "xml_id": "shopfloor.packing_info_{}".format(xml_id),
```

Comment:
this xid is wrong. Also, I wouldn't add any xid because you did not have any record before, just strings inserted manually,

Reply:
You are right, the prefix should be ```shopfloor_packing_info```. The XML-id is placed here to ease exporting the existing records to clean them and then importing them in several databases without having to depend on the XML-id assigned by each database. Right now, this is actually not needed, and is thinking about the future when we do this field translatable. If you still don't want it, I can remove it for now...

Reply:
IMO it's useless. If you keep it I'd replace the module name w/ `__export__`

Reply:
Doesn't matter, I'll remove it.